misk/src/main/kotlin/misk/web/interceptors/RebalancingInterceptor.kt

```diff
@@ -0,0 +1,57 @@
+package misk.web.interceptors
+
+import misk.Action
+import misk.logging.getLogger
+import misk.random.ThreadLocalRandom
+import misk.web.NetworkChain
+import misk.web.NetworkInterceptor
+import misk.web.Response
+import misk.web.WebConfig
+import javax.inject.Inject
+import javax.inject.Singleton
+
+private val logger = getLogger<RebalancingInterceptor>()
+
+/**
+ * When we're deploying and redeploying our pods, we want to make sure that clients rebalance onto
+ * the new pods. This randomly closes connections so they will be recreated, naturally balancing
+ * connections across pods.
+ */
+class RebalancingInterceptor @Inject constructor(
+  private val random: ThreadLocalRandom,
+  private val probability: Double
+) : NetworkInterceptor {
+  override fun intercept(chain: NetworkChain): Response<*> {
+    val response = chain.proceed(chain.request)
+
+    if (random.current().nextDouble() < probability) {
+      return response.copy(headers = response.headers.newBuilder()
+          .set("Connection", "close")
```

Comment:
Does `"Connection": "close"` translate to `GOAWAY` in Jetty? `"Connection": "close"` isn't part of the http2 protocol. The Go http2 server explicitly added it at some point. Jetty doesn't seem to have similar support: https://github.com/eclipse/jetty.project/issues/2788